src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -1328,29 +1249,188 @@ export default class DownloadTab extends React.Component<
+                                    sample.studyId
+                                ].map(profile => {
+                                    return {
+                                        molecularProfileId:
+                                            profile.molecularProfileId,
+                                        sampleId: sample.sampleId,
+                                    } as SampleMolecularIdentifier;
+                                })
+                            );
+                            return acc;
+                        }, []);
+
+                    if (sampleIdentifiers.length) {
+                        molecularData = await client.fetchMolecularDataInMultipleMolecularProfilesUsingPOST(
+                            {
+                                projection:
+                                    REQUEST_ARG_ENUM.PROJECTION_DETAILED,
+                                molecularDataMultipleStudyFilter: {
+                                    entrezGeneIds: _.map(
+                                        genes,
+                                        (gene: Gene) => gene.entrezGeneId
+                                    ),
+                                    sampleMolecularIdentifiers: sampleIdentifiers,
+                                } as MolecularDataMultipleStudyFilter,
+                            }
+                        );
+                    }
+                }
+
+                // STEP 2: generate alteration data
+                const data = {
+                    samples: _.groupBy(
+                        molecularData,
+                        data => data.uniqueSampleKey
+                    ),
+                } as CaseAggregatedData<ExtendedAlteration>;
+
+                const alterationData = generateOtherMolecularProfileData(
+                    selectedProfiles.map(profile => profile.molecularProfileId),
+                    data
+                );
+
+                // STEP 3: generate download data
+                const downloadData = generateOtherMolecularProfileDownloadData(
+                    alterationData,
+                    samples,
+                    genes
+                );
+
+                // STEP 4: download data
+                fileDownload(
+                    this.downloadDataText(downloadData),
+                    `${profileName}.txt`
                 );
-                fileDownload(textMap[profileName], `${profileName}.txt`);
             }
         );
     }
 
     @autobind
     private handleTransposedOtherMolecularProfileDownload(profileName: string) {
-        onMobxPromise(
-            this.allOtherMolecularProfileDownloadDataGroupByProfileName,
-            downloadDataGroupByProfileName => {
-                const transposedTextMap = this.downloadDataTextGroupByKey(
-                    this.unzipDownloadDataGroupByKey(
-                        downloadDataGroupByProfileName
-                    )
+        onMobxPromise<any>(
```

Comment:
maybe put a comment above this explaining what "other" means.  i think it means non-selected, right?

Reply:
I changed those `Other` to `NonSelected`